includes/acl/sjson/sjson_parser.h

```diff
@@ -0,0 +1,719 @@
+#pragma once
+
+////////////////////////////////////////////////////////////////////////////////
+// The MIT License (MIT)
+//
+// Copyright (c) 2017 Nicholas Frechette & Animation Compression Library contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+////////////////////////////////////////////////////////////////////////////////
+
+#include "sjson_parser_error.h"
+
+#include <cctype>
+#include <cmath>
+#include <cstring>
+
+namespace acl
+{
+	class SJSONParser
+	{
+	public:
+		SJSONParser(const char* const input, int input_length)
+			: m_input(input),
+			  m_input_length(input_length)
+		{
+			init_state();
+		}
+
+		bool object_begins()
+		{
+			return read_opening_brace();
+		}
+
+		bool object_begins(const char* const having_name)
```

Comment:
not sure the double const is really required since we don't use const for arguments in general when they aren't pointers, just const char* having_name would be good enough IMO, idem elsewhere